packages/web-forms/src/components/common/map/vertex-geometry.ts

```diff
@@ -0,0 +1,228 @@
+import type { Coordinate } from 'ol/coordinate';
+import Feature from 'ol/Feature';
+import { Point, LineString, Polygon } from 'ol/geom';
+
+export const getFlatCoordinates = (geometry: LineString | Polygon | undefined) => {
+	if (!geometry) {
+		return [];
+	}
+
+	if (geometry instanceof LineString) {
+		return geometry.getCoordinates();
+	}
+
+	return geometry.getCoordinates()[0] ?? [];
+};
+
+const getDeltaBetweenPoints = (pointA: number[], pointB: number[]) => {
+	const isDefined = (c: number) => c != null;
+	const pairA = pointA.filter(isDefined) as [number, number];
+	const pairB = pointB.filter(isDefined) as [number, number];
+	if (pairA.length < 2 || pairB.length < 2) {
+		return { deltaX: 0, deltaY: 0 };
+	}
```

Comment:
Filtering this feels strange because if for some reason `x` were null then y would become x and altitude would become y, right?

If so I think it'd be cleaner to leave the coordinates where they are and abort as soon as either a or b is null in index 0 or 1.

eg:


```suggestion
	const isDefined = (point: number[]) => point.length > 1 && point[0] && point[1];
	if (!isDefined(pointA) || !isDefined(pointB)) {
		return { deltaX: 0, deltaY: 0 };
	}
```